sources/core/Stride.Core.CompilerServices.Tests/CompilerUtils.cs

```diff
@@ -0,0 +1,89 @@
+            return (compilation, diagnostics);
+        }
+
+        private static Compilation CreateCompilation(string assemblyName, string source)
+            => CSharpCompilation.Create(assemblyName,
+                new[] { CSharpSyntaxTree.ParseText(source) },
+                new[]
+                {
+                    // System.Runtime.dll
+                    MetadataReference.CreateFromFile(typeof(object).GetTypeInfo().Assembly.Location.Replace("Private.CoreLib", "Runtime")),
+                    // System.Private.CoreLib.dll
+                    MetadataReference.CreateFromFile(typeof(object).GetTypeInfo().Assembly.Location),
+                    // Stride.Core.dll
+                    MetadataReference.CreateFromFile(typeof(object).GetTypeInfo().Assembly.Location.Replace("System.Private.CoreLib", "Stride.Core")),
+                },
+                new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary));
+        public static DiagnosticAnalyzer[] AllAnalyzers => typeof(DiagnosticsAnalyzerHelper).Assembly.GetTypes()
+            .Where(t => t.IsSubclassOf(typeof(DiagnosticAnalyzer)) && !t.IsAbstract)
+            .Select(type => (DiagnosticAnalyzer)Activator.CreateInstance(type)).ToArray();
+
+        public static Compilation CreateCompilation(string sourceCode)
+        {
+            // Create a syntax tree from the source code
+            SyntaxTree syntaxTree = CSharpSyntaxTree.ParseText(SourceText.From(sourceCode));
+
+            // Create a compilation with your syntax tree and necessary references
+            MetadataReference[] references = new[]
+            {
+                // System.Runtime.dll
+                MetadataReference.CreateFromFile(typeof(object).GetTypeInfo().Assembly.Location.Replace("Private.CoreLib", "Runtime")),
+                // System.Private.CoreLib.dll
+                MetadataReference.CreateFromFile(typeof(object).GetTypeInfo().Assembly.Location),
+                // Stride.Core.dll
+                MetadataReference.CreateFromFile(typeof(object).GetTypeInfo().Assembly.Location.Replace("System.Private.CoreLib", "Stride.Core")),
+
+            };
+
+            CSharpCompilation compilation = CSharpCompilation.Create(
+                "TestAssembly",
+                syntaxTrees: new[] { syntaxTree },
+                references: references,
+                options: new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary));
+
+            return compilation;
+        }
+
+        public static ImmutableArray<Diagnostic> GetAnalyzerDiagnostics(
+            this Compilation compilation,
+            params DiagnosticAnalyzer[] analyzers)
+        {
+            // Analyze the compilation and get diagnostics
+            var compilationWithAnalyzers = compilation.WithAnalyzers(ImmutableArray.Create(analyzers));
+            var diagnostics = compilationWithAnalyzers.GetAnalyzerDiagnosticsAsync().Result;
```

Comment:
the tests are pretty fast allready, i doubt that async is needed

should i search for async even though its probably not needed at all?

Reply:
It's not about speed, but more about avoiding threading issues. I think it may be fine for this case when used in the tests, but the general rule is to avoid `.Result` and use the `async/await` pattern instead. Let's leave it as is.